benchmarks/framework_comparison/random_walks/launcher.py

```diff
@@ -0,0 +1,92 @@
+import platform
+import timeit
+from importlib.metadata import version
+
+import mlx.core as mx
+import numpy as np
+import torch
+import torch_geometric.datasets as pyg_datasets
+from setup_mlx import mlx_random_walks
+from setup_numpy import random_walks_numpy
+from setup_torch_cluster import torch_cluster_random_walk
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from utils import to_markdown_table
+
+import mlx_graphs.datasets as mxg_datasets
+
+mx.set_default_device(mx.gpu)
+
+torch.manual_seed(42)
+mx.random.seed(42)
+np.random.seed(42)
+
+FRAMEWORKS = [
+    "mxg",
+    "pyg",
+    "numpy",
+]
+
+frameworks_to_benchmark = {
+    "mxg": mlx_random_walks,
+    "pyg": torch_cluster_random_walk,
+    "numpy": random_walks_numpy,
+}
+
+frameworks_to_dataset = {
+    "mxg": mxg_datasets.PlanetoidDataset(name="cora", base_dir="~"),
+    "pyg": pyg_datasets.Planetoid(root="data/Cora", name="Cora"),
+    "numpy": pyg_datasets.Planetoid(root="data/Cora", name="Cora"),
+}
+COMPILE = {"mxg": [False, True], "pyg": [False], "numpy": [False]}
+
+
+def benchmark(framework, edge_index, start_indices, walk_length, compile=None):
+    randomwalks = frameworks_to_benchmark[framework]
+    randomwalks(edge_index, start_indices, walk_length, compile)
+
+
+loader = DataLoader(range(2708), batch_size=2000)
+start_indices = next(iter(loader))
+
+results = [["Framework", "Time/epoch"]]
+for framework in tqdm(FRAMEWORKS):
+    for compile in COMPILE[framework]:
+        dataset = frameworks_to_dataset[framework]
+
+        if framework == "mxg":
+            edge_index = dataset.graphs[0].edge_index
+        else:
+            edge_index = dataset.edge_index
+
+        times = timeit.Timer(
+            lambda: benchmark(
+                framework,
+                edge_index=edge_index,
+                start_indices=start_indices,
+                walk_length=10000,
+                compile=compile,
+            )
+        ).repeat(repeat=1, number=1)
```

Comment:
This generates just one me sample, you may want to increase the `repeat` number to get the best out of (say) 5 runs